google/cloud/pubsub/internal/tracing_message_batch.cc

```diff
@@ -12,38 +12,30 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-#ifdef GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY
-
 #include "google/cloud/pubsub/internal/tracing_message_batch.h"
 #include "google/cloud/pubsub/internal/message_batch.h"
-#include "google/cloud/pubsub/version.h"
+#include "google/cloud/version.h"
```

Comment:
I think we could simplify this to:

```cc
#include "google/cloud/pubsub/internal/tracing_message_batch.h"
#if GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY_H
... all the necessary #includes here ...

namespace google {
namespace cloud {
namespace pubsub_internal {
GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN

GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_END
}  // namespace pubsub_internal
}  // namespace cloud
}  // namespace google

#else

namespace google {
namespace cloud {
namespace pubsub_internal {
GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN

std::shared_ptr<MessageBatch> MakeTracingMessageBatch(
    std::shared_ptr<MessageBatch> message_batch) {
  return message_batch;
}
GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_END
}  // namespace pubsub_internal
}  // namespace cloud
}  // namespace google

#endif  // GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY

Reply:
Or maybe:

I think we could simplify this to:

```cc
#include "google/cloud/pubsub/internal/tracing_message_batch.h"
#if GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY_H
... all the necessary #includes here ...
#endif 

namespace google {
namespace cloud {
namespace pubsub_internal {
GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN

#if GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY_H

// ... all the code here ...

#else

std::shared_ptr<MessageBatch> MakeTracingMessageBatch(
    std::shared_ptr<MessageBatch> message_batch) {
  return message_batch;
}

#endif  // GOOGLE_CLOUD_CPP_HAVE_OPENTELEMETRY

GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_END
}  // namespace pubsub_internal
}  // namespace cloud
}  // namespace google

```

Reply:
I think the second one is what I have. Am I missing something?

Reply:
Oh i guess I could move the includes inside the otel ifdef, but those are includes based on the non-otel code.

Reply:
In the non-OTel case I expect `#include "google/cloud/pubsub/internal/tracing_message_batch.h"` provides all the types you need.  If it does not then there is a bug in the header.  But maybe IWYU is forcing you to add even more redundant includes than I thought it would.

Reply:
Removed. 

I'm not sure if IWYU will complain or not, so 🤷